### PR TITLE
Add foxglove_bridge package

### PR DESCRIPTION
**Pull request by defunctzombie**
The foxglove bridge package is a c++ websocket bridge for connecting ROS systems to Foxglove Studio. It is similar in spirit to rosbridge but written in c++ to overcome some performance problems we experienced with rosbridge under large message or high rate message workloads.

https://foxglove.dev/blog/announcing-the-new-foxglove-bridge-for-live-ros-data
https://github.com/foxglove/ros-foxglove-bridge/

Humble PR: https://github.com/RoboStack/ros-humble/pull/49

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
Thanks for the contribution @defunctzombie!

Unfortunately our CI seems to have some problems unrelated to this package. Any idea what's going on here @wolfv? Seems like some boa issue:
```
Recipe validation error
[124](https://github.com/RoboStack/ros-noetic/actions/runs/4247326654/jobs/7385241857#step:15:125)

[125](https://github.com/RoboStack/ros-noetic/actions/runs/4247326654/jobs/7385241857#step:15:126)
Additional properties are not allowed ('outputs' was unexpected)
[126](https://github.com/RoboStack/ros-noetic/actions/runs/4247326654/jobs/7385241857#step:15:127)
```

**Comment by Tobias-Fischer** [comment]
Ah, actually it might be related to the package after all. Has this been released in the ROS build farm yet?

**Comment by defunctzombie** [comment]
@Tobias-Fischer I got the name wrong. I used the repo name rather than the package name. I've updated the PR with the correct name.

**Comment by defunctzombie** [comment]
I don't expect this PR and the humble one will pass the builds on the first try. We might need to adjust some aspects of our packaging or build.

**Comment by Tobias-Fischer** [comment]
Ok cool - now fails with a less cryptic error:
```
$PREFIX/include/actionlib/managed_list.h:221:73: error: '_1' was not declared in this scope
[2020](https://github.com/RoboStack/ros-noetic/actions/runs/4247483049/jobs/7385580850#step:14:2021)
  221 |     return add(elem, boost::bind(&ManagedList<T>::defaultDeleter, this, _1) );
[2021](https://github.com/RoboStack/ros-noetic/actions/runs/4247483049/jobs/7385580850#step:14:2022)
      |                                                                         ^~
[2022](https://github.com/RoboStack/ros-noetic/actions/runs/4247483049/jobs/7385580850#step:14:2023)
```

This needs std::place_holders or sth like that. Let me know if you run into any issues that you can't seem to solve.

**Comment by achim-k** [comment]
The error stems from `actionlib`:

```
$PREFIX/include/actionlib/managed_list.h: In member function 'actionlib::ManagedList<T>::Handle actionlib::ManagedList<T>::add(const T&)':
$PREFIX/include/actionlib/managed_list.h:221:73: error: '_1' was not declared in this scope
  221 |     return add(elem, boost::bind(&ManagedList<T>::defaultDeleter, this, _1) );
      |                                                                         ^~
```

A few lines above it says that declaring boost placeholders in the global namespace is deprecated:
```
$PREFIX/include/boost/bind.hpp:36:1: note: '#pragma message: The practice of declaring the Bind placeholders (_1, _2, ...) in the global namespace is deprecated. Please use <boost/bind/bind.hpp> + using namespace boost::placeholders, or define BOOST_BIND_GLOBAL_PLACEHOLDERS to retain the current behavior.'
```

Is `foxglove_bridge` the first package that (transitively) depends on `actionlib`?

**Comment by Tobias-Fischer** [comment]
Whatever is easier for you :). You can easily patch packages - see https://github.com/RoboStack/ros-noetic/tree/master/patch

File: vinca_win.yaml [hunk 1]
```diff
@@ -75,6 +75,7 @@ packages_select_by_deps:
   - pcl-ros
   - pcl-conversions
   - velodyne-simulator
+  - foxglove_bridge
```
Comment (defunctzombie) [inline vinca_win.yaml:78]:
```suggestion
```

**Comment by defunctzombie** [comment]
@achim-k Seems like there are some more instances of this issue in the `ros-noetic-ros-babel-fish/` package.

**Comment by achim-k** [comment]
> @achim-k Seems like there are some more instances of this issue in the `ros-noetic-ros-babel-fish/` package.

No, it's still from `actionlib`. Apparently adding the patch in https://github.com/RoboStack/ros-noetic/pull/345/commits/9bb189938c51afb0e8aea9f8aebb00c5319c2557 didn't work

File: vinca_linux_64.yaml [hunk 2]
```diff
@@ -104,6 +104,7 @@ packages_select_by_deps:
   # - rosmon
   - catch-ros
   - rosfmt
+  - foxglove_bridge
```
Comment (defunctzombie) [inline vinca_linux_64.yaml:107]:
```suggestion
  - foxglove_bridge

```

**Comment by defunctzombie** [comment]
@achim-k I was looking at the linux-64 build. I figure until we get that one fixed the others have less hope.

**Comment by Tobias-Fischer** [comment]
See my PR to your repo which will build all selected packages (so you can rebuild actionlib etc. if needed). Many thanks again for your contributions!

**Comment by defunctzombie** [comment]
@Tobias-Fischer FYI - this change also commented out all the other packages since we were iterating on this and didn't need to keep building those. Probably you didn't mean to also merge those changes?

**Comment by Tobias-Fischer** [comment]
No, this was deliberate, all good :)

Thanks again for the contribution!

**Pull request merged**